tests/lib/cli.js

```diff
@@ -858,6 +858,148 @@ describe("cli", () => {
 
     });
 
+    describe("when passed --fix-dry-run", () => {
+        const sandbox = sinon.sandbox.create();
```

Comment:
I'm not a big fan of using stubs in tests, since they tend to make the tests fragile ([explanation](https://github.com/airbnb/javascript/blob/f1cc91c88a9feb9d120a61897c2285821afea634/README.md#testing--for-real)). For example, in this case `CLIEngine` makes no guarantees about whether it calls `outputFixes` internally in fix mode -- it only makes guarantees about whether it touches the filesystem. This test makes assertions about whether `outputFixes` is called, so it's possible that it would break or fail to catch bugs later on (since it's testing an implementation detail, rather than the spec).

It looks like this file already contains a bunch of tests with stubs, so I don't really have an objection to adding a few more (they're better than nothing). However, I think it would be good to also add a few integration tests for this behavior in [`tests/bin/eslint.js`](https://github.com/eslint/eslint/blob/ffa021e7696b24722cbbef306d494dd64d7b5215/tests/bin/eslint.js). For example, a test could pipe code to `eslint --stdin --fix-dry-run` and assert that the result contains messages and the output. Another test could run `eslint --fix-dry-run` on a temporary file and assert that the messages/output are correct, but the contents of the file aren't updated.

Reply:
Ok, I'll add the integration tests then.